api/envoy/config/filter/accesslog/v2/accesslog.proto

```diff
@@ -188,3 +191,30 @@ message ResponseFlagFilter {
     ]
   }];
 }
+
+// Filters gRPC requests based on their response status. If a gRPC status is not provided, the
+// filter will infer the status from the HTTP status code.
+message GrpcStatusFilter {
+  // Logs only responses that have any one of the gRPC statuses in this field.
+  repeated string statuses = 1 [(validate.rules).repeated .items.string = {
+    in: [
```

Comment:
define an enum instead of use PGV to limit the name?

Reply:
I was following the structure in ResponseFlagFilter, but I like using an enum instead.

Could we even supplant the GrpcStatus enum in `//include/envoy/grpc/status.h` with a proto, and use that as the source of truth? It would ensure that this proto always stays in sync with available gRPC statuses (if the spec should change), and prevent code duplication across protobuf / C++.

Reply:
Not sure about that, the status is already copied across many places (protobuf, grpc, etc) though.

Reply:
Gotcha. I'll stick to just changing this to an enum for this PR, and I can look into the feasibility of a switch to a single enum afterward.

Reply:
I got a chance to look at this more closely today. I think changing from strings to enum would set off a bit of a chain reaction in terms of design, and I just want to get a second opinion on it:

Changing the string type to an enum type would entail that `nameToGrpcStatus` would have to be changed to use the relevant enum type. Because that enum only exists in the `GrpcStatusFilter` proto, it doesn't make sense to have that function inside of `Grpc::Utility` anymore. Therefore the functionality should exist within `GrpcStatusFilter`, but if it were a private method then it wouldn't be as explicitly testable, and if it were a public function then the filter would export functionality unrelated to its core purpose.

I'm thinking this leaves two real options:

1. Opt for explicit testability, and add the conversion to `GrpcStatusFilter` as a public function, along with a test.
2. Opt for cleaner class structure, and implicitly test the correctness of the conversion via more rigorous testing on the filter passing / blocking.

I'm leaning towards option 2, but I want to hear what you think.